docs/spec/spec.md

```diff
@@ -0,0 +1,1380 @@
+
+###### Example: Hierarchical Resource Path
+
+```ballerina
+service graphql:Service on new graphql:Listener(4000) {
+    resource function get profile/address/number() returns int {
+        return 308;
+    }
+
+    resource function get profile/address/street() returns string {
+        return "Negra Arroyo Lane";
+    }
+
+    resource function get profile/address/city() returns string {
+        return "Albuquerque";
+    }
+
+    resource function get profile/name() returns string {
+        return "Walter White";
+    }
+
+    resource function get profile/age() returns int {
+        return 52;
+    }
+}
+```
+
+The above example shows how to use hierarchical resource paths to create a hierarchical data model. When the schema is generated using this service, the root `Query` operation has a single field, `profile`, as it is the only path segment at the top level. The type of this field is also `profile`, which is an `Object` type. This object type has three fields: `address`, `name`, and `age`. The type of the `address` field is also `Object` as it is an intermediate path segment (i.e. has child path segments). The name of this object type is `address`. It has three fields: the `number` (type `Int!`), the `street` (type `String`), and the `city` (type `String`). The `name` field is of type `String!`, and the `age` field is of type `Int!`. Check [Types Section](#4-types) for more information on types and fields.
+
+### 3.4 Remote Methods
+
+The `remote` methods are used to define the fields of the `Mutation` type in a GraphQL schema. Remote methods are validated at the compile-time.
+
+> **Note:** The `resource` and `remote` methods are called __*resolver functions*__ in GraphQL terminology. Therefore, in this spec, sometimes the term __*resolver function*__ is used to refer `resource` and `remote` methods.
+
+#### 3.4.1 Remote Method Name
+
+The name of the `remote` method is the name of the corresponding GraphQL field in the `Mutation` type.
+
+## 4. Types
+
+GraphQL type system is represented using a hierarchical structure. Type is the fundamental unit of any GraphQL schema.
+
+### 4.1 Scalars
+
+Scalar types represent primitive leaf values in the GraphQL type system. The following built-in types are supported in the Ballerina GraphQL package. Scalar values are represented by the primitive types in Ballerina.
+
+#### 4.1.1 Int
+The `Int` type is represented using the `int` type in Ballerina.
+
+#### 4.1.2 Float
+The `Float` type is represented using the `float` type in Ballerina.
+
+> **Note:** When used as an input value type, both integer and float values are accepted as valid inputs.
```

Comment:
```suggestion
> **Note:** When `Float` used as an input value type, both integer and float values are accepted as valid inputs.
```

Reply:
Since the note is under the `Float` type section, do we need to add this?